### ISSUE TITLE
Very long project names cause global menu to cut off

### ISSUE THREAD

**Issue by davidnuzik** [rancher/rancher#17381]
Version: master 2.2 (1/10/19)

**What kind of request is this (question/bug/enhancement/feature request):**
Bug, minor, requires unusually long project names

**Steps to reproduce (least amount of steps as possible):**
- When extremely long project names exist, the page styling won't adjust the width of the menu properly when the browser window is rather wide (such as nearly 1920px). 

**Result:**
This results in an issue where you can no longer see the vertical scrollbar. It is fixed when the browser window width is decreased manually. If you have project names with crazy long names (for example 5000 chars) the menu doesn't work anymore and you cannot see all the clusters as easily (second screenshot)
<img width="1474" alt="in_some_cases_menu_not_adjusting_to_browser_dimensions" src="https://user-images.githubusercontent.com/45179589/49105070-ea707c00-f23c-11e8-9c89-e7c6bfaccf03.png">
<img width="916" alt="screen shot 2018-11-27 at 9 33 35 am" src="https://user-images.githubusercontent.com/45179589/49108251-f2341e80-f244-11e8-8975-59917de2b58b.png">

**Comment by davidnuzik** [comment]
Suggestion: Perhaps we can just put "..." at the end for cluster/project names when they approach the width of the nav bar. (and that should keep gloabal menu width under control I think)

**Comment by davidnuzik** [comment]
Version: Master (v2.3) (5/20/19)

The global menu now has a limited width. Even when I make the browser window rather small almost to where it will switch to mobile view it will shrink down in width appropriately.

I tested various window dimensions and project name lengths and didn't run into any issues.

Below are a couple of screenshots just showing one of the project names I tested with. The name is cut off at a reasonable length (pretty long) and a "..." is used to truncate
![image](https://user-images.githubusercontent.com/45179589/58037723-e8e48a00-7ae2-11e9-9268-fd22cc4ff350.png)

![image](https://user-images.githubusercontent.com/45179589/58037737-ee41d480-7ae2-11e9-898b-d8feaf29d09e.png)
